File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& v) {
        sort(v.begin(),v.end());
        vector<vector<int>>ans;
        ans.push_back(v[0]);
        for(int i=1;i<v.size();i++){
            if(v[i][0]<=ans.back()[1]){
                ans.back()[1]=max(v[i][1],ans.back()[1]);
            }else{
                ans.push_back(v[i]);
            }
        }
        return ans;
    }
};

File: 56-merge-intervals/merge-intervals.py
class Solution:
    def merge(self, nums: List[List[int]]) -> List[List[int]]:
        l=[]
        nums.sort()
        l.append(nums[0])

        for i in range(len(nums)):
            if l[-1][1] >=nums[i][0]:
                l[-1][1]=max(l[-1][1],nums[i][1])
            else:
                l.append(nums[i])
        return l
